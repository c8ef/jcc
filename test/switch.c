int main() {
  int x = 42;
  switch (x) {
    case 42:
      return 1;
    default:
      return 2;
  }
}
