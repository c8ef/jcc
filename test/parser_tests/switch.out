FunctionDecl: main
  Args[0]
  CompoundStatement
    DeclStatement
      VarDecl: x
        IntergerLiteral: 42
    SwitchStatement
      DeclRefExpr: x
      CompoundStatement
        CaseStatement
        value: 42
          ReturnStatement
            IntergerLiteral: 1
        DefaultStatement
          ReturnStatement
            IntergerLiteral: 2
